lms/services/user.py

```diff
@@ -175,10 +174,8 @@ def get_users_for_assignment(
             )
             .where(
                 LMSUserAssignmentMembership.assignment_id == assignment_id,
-                LMSUserAssignmentMembership.lti_role_id.in_(
-                    select(LTIRole.id).where(
-                        LTIRole.scope == role_scope, LTIRole.type == role_type
-                    )
+                self._filter_membership_by_role_clause(
```

Comment:
Adding a small helper function here to avoid a bit duplication.

Again, this will make more sense once we add `exclude_role`, the helper function will do the filtering for both.